tests/backup/backup_share_test.go

```diff
@@ -1664,15 +1664,37 @@ var _ = Describe("{CancelClusterBackupShare}", func() {
 				log.FailOnError(err, "Failed to get all backups for user - [%s]", chosenUser)
 				log.Infof("Backups user [%s] has access to - %v", chosenUser, userBackups)
 				if len(userBackups) > 0 {
-					return "", true, fmt.Errorf("waiting for all backup access - [%v] to be revoked for user = [%s]",
-						userBackups, chosenUser)
+					userBackups, err = GetAllBackupsForUser(chosenUser, commonPassword)
+					log.FailOnError(err, "Failed to get all backups for user - [%s]", chosenUser)
+					ctx, err := backup.GetAdminCtxFromSecret()
+					log.FailOnError(err, "Fetching px-central-admin ctx")
+					for _, backupName := range userBackups {
+						backupUID, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+						log.FailOnError(err, fmt.Sprintf("Getting UID for backup %v", backupName))
+						backupInspectRequest := &api.BackupInspectRequest{
+							Name:  backupName,
+							Uid:   backupUID,
+							OrgId: orgID,
+						}
+						resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+						log.FailOnError(err, fmt.Sprintf("error inspecting backup %v", backupName))
+						deletePendingStatus := api.BackupInfo_StatusInfo_DeletePending
+						deletingStatus := api.BackupInfo_StatusInfo_Deleting
+						actual := resp.GetBackup().GetStatus().Status
+						reason := resp.GetBackup().GetStatus().Reason
+						if actual == deletePendingStatus || actual == deletingStatus {
+							log.Infof("Ignoring the backup from user list as backup is in [%s] state ,Reason:[%s]", actual, reason)
+							return "", false, nil
+						} else {
+							return "", true, fmt.Errorf("waiting for all backup access - [%v] to be revoked for user = [%s]",
+								userBackups, chosenUser)
+						}
+					}
 				}
 				return "", false, nil
 			}
 			_, err = task.DoRetryWithTimeout(noAccessCheck, 5*time.Minute, 30*time.Second)
```

Comment:
This accesscheck is specific to this testcase level.

Reply:
reply to (https://github.com/portworx/torpedo/pull/1301#discussion_r1161401653):
But you can still have `noAccessCheckTimeout` and `noAccessCheckRetryDuration` at the testcase level variable, as you are using the values **twice**, and it will be easier to change in variable definitions (under `Describe()`).

Reply:
Done sir.